platform/core/src/services/ToolBarService/ToolbarService.ts

```diff
@@ -94,8 +94,9 @@ export default class ToolbarService extends PubSubService {
             commandsManager.runCommand(commandName, commandOptions, context);
           });
 
-          // only set the primary tool if no error was thrown
-          this.state.primaryToolId = itemId;
+          // only set the primary tool if no error was thrown and if the itemId is not
+          // undefined set the first tool as the primary tool
```

Comment:
It is not clear from the comment why we are setting the primaryToolId the way we are. Why might `itemId` be undefined? Also the comment suggests that the first tool is used if `itemId` is not undefined.

Reply:
for backward compatibility i put it like this, we don't really need itemId